### PR TITLE
support partial file downloads for exports

### PR DESCRIPTION
**Pull request by snopoke**
This should allow failed downloads to be resumed.

https://manage.dimagi.com/default.asp?277653

```
# download first 10000 bytes
curl -n <export url> -u username -o export.zip -r 0-10000

# download rest
curl -n <export url> -u username -o export.zip -C -
```

### CODE REVIEW

File: corehq/util/download.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    """
+    :param payload: File like object.
+    :param content_length: Size of payload in bytes
+    :param content_format: ``couchexport.models.Format`` instance
+    :param filename: Name of the download
+    :param request: The request. Used to determine if a range response should be given.
+    :return: HTTP response
+    """
+    ranges = None
+    if "HTTP_RANGE" in request.META:
```
Comment (pr33thi) [inline corehq/util/download.py]: I believe this will error if no `request` is passed in and the default value of `request=None` is used

**Review by pr33thi — CHANGES_REQUESTED** [review]
Just found the 1 bug, besides that it looks good

**Review by pr33thi — APPROVED** [review]

**Comment by dannyroberts** [comment]
Seems to be causing problems because `werkzeug` isn't in `requirements.txt` (it's in `dev-requirements.txt`. That said, I can't for the life of me figure out why tests didn't catch that. Running tests locally without `werkzeug` installed gives me an import error

```
...
    from werkzeug.http import parse_range_header
ImportError: No module named werkzeug.http
```

**Comment by pr33thi** [comment]
@dannyroberts  @proteusvacuum pointed out that dev-requirements.txt is installed in docker, so it's used in travis:
https://github.com/dimagi/commcare-hq/blob/master/Dockerfile#L29

I'm making a pr to move it from dev-requirements to requirements.txt

**Comment by proteusvacuum** [comment]
Also: https://github.com/dimagi/commcare-hq/pull/20930

**Comment by snopoke** [comment]
Shucks, sorry about that.

**Pull request merged**